pages/resources/[page].js

```diff
@@ -1,75 +1,254 @@
+import Form from 'components/Form/Form';
+import Input from 'components/Form/Input/Input';
+import Button from '../../components/Button/Button';
 import styles from '../styles/resources.module.css';
+import Select from '../../components/Form/Select/Select';
+import Alert from '../../components/Alert/Alert';
+import { RESOURCE_SEARCH } from '../../common/constants/testIDs';
 
 const pageTitle = 'Resources';
 
-function Resources() {
+const ResourceCard = dynamic(() => import('../../components/Cards/ResourceCard/ResourceCard'), {
+  loading: () => <ResourceSkeletonCard numberOfSkeletons={5} />,
+});
+
+Resources.propTypes = {
+  initialValues: shape({
+    category: string,
+    q: string,
+    languages: oneOfType([string, array]),
+    paid: string,
+  }),
+};
+
+Resources.defaultProps = {
+  initialValues: {
+    category: '',
+    q: '',
+    languages: [],
+    paid: '',
+  },
+};
+
+function Resources({ initialValues }) {
   const router = useRouter();
   const { pathname, query } = router;
-  const currentPage = parseInt(query.page, 10);
+  const { page, category, languages, paid, q } = query;
+  const currentPage = parseInt(page, 10);
 
-  if (!isNumber(currentPage)) {
-    // TODO: Handle situation where user tried some funny business, like `/resources/fuck-you`
-    throw new Error(`'currentPage' is not a number`);
+  // eslint-disable-next-line no-restricted-globals
+  if (page && isNaN(page)) {
+    router.push({ pathname: pathname.replace('[page]', '1') });
   }
+  const [errorMessage, setErrorMessage] = useState(null);
 
-  const [isLoading, setIsLoading] = useState(true);
   const [resources, setResources] = useState([]);
   const [totalPages, setTotalPages] = useState(currentPage);
 
+  const [allCategories, setAllCategories] = useState([]);
+  const [allLanguages, setAllLanguages] = useState([]);
+
+  const costOptions = [
+    { label: 'Paid', value: 'true' },
+    { label: 'Free', value: 'false' },
+  ];
+
+  const handleError = error =>
+    error.errors
+      ? error.errors['not-found'].message
+      : 'There was a problem loading those resources.';
+
+  const handleEndpoint = () => {
+    if (q) {
+      return getResourcesBySearch({ page, category, languages, paid, q });
+    }
+    return getResourcesPromise({ page, category, languages, paid });
+  };
+
+  const handleSubmit = (values, actions) => {
+    const emptyQueryParameters = Object.entries(values).filter(
+      item => item[1] === null || !item[1].length,
+    );
+    const activeParameters = omit(
+      values,
+      emptyQueryParameters.map(parameter => parameter[0]),
+    );
+
+    updateQuery(activeParameters);
+    setTimeout(() => {
+      actions.setSubmitting(false);
+    }, 500);
+  };
+
   useEffect(() => {
-    getResourcesPromise({ page: query.page })
+    Promise.all([getResourcesByCategories(), getResourcesByLanguages()])
+      .then(([categoriesResponse, languagesResponse]) => {
+        const {
+          data: { data: categoriesData },
```

Comment:
It might be useful to bring up the weirdness of `data.data` to Aaron. If we keep on adding more stuff it's only gonna get more obtuse.